src/Lachain.Consensus/HoneyBadger/HoneyBadger.cs

```diff
@@ -136,6 +138,19 @@ private void HandleCommonSubset(ProtocolResult<CommonSubsetId, ISet<EncryptedSha
                 var dec = _privateKey.Decrypt(share);
                 _taken[share.Id] = true;
                 _receivedShares[share.Id] = share;
+                if (_decryptedShares[share.Id].Count > 0) // if we have any partially decrypted shares for this share - verify them
+                {
+                    if (!_skipDecryptedShareValidation)
+                    {
+                        if (Wallet.GetTpkeVerificationKey(share.Id) is null)
+                            _decryptedShares[share.Id].Clear();
+                        else
+                            _decryptedShares[share.Id] = _decryptedShares[share.Id]
+                                .Where(ps => Wallet.GetTpkeVerificationKey(share.Id)!.VerifyShare(share, ps))
```

Comment:
I think it should be Wallet.GetTpkeVerificationKey(ps.DecryptorId) to get the public key of decryptor

Reply:
You are right,  thank you